apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -1426,6 +1490,14 @@ maybe_reply(Actions, From, Reply) ->
 
 -spec data_record_to_external_map(data()) -> resource_data().
 data_record_to_external_map(Data) ->
+    AddedChannelsList = maps:to_list(Data#data.added_channels),
+    AddedChannelsListWithoutConfigs =
+        [
+            {ChanID, maps:remove(config, Status)}
+         || {ChanID, Status} <- AddedChannelsList
+        ],
+    AddedChannelsWithoutConfigs =
+        maps:from_list(AddedChannelsListWithoutConfigs),
```

Comment:
Nit:

```suggestion
    AddedChannelsWithoutConfigs =
        maps:map(
            fun(_ChanID, Status) -> maps:remove(config, Status) end,
            Data#data.added_channels
        ),
```

Reply:
Fixed [88c96e2](https://github.com/emqx/emqx/pull/13077/commits/88c96e26de41199f4fb66f69be2960a04511098f)

Also added another fix to not leak configs in alarm.